README.md

```diff
@@ -22,88 +25,125 @@ def backend():
 
-#### local_git_repo(path)
-Creates a `repo` with the content at `path`.
+#### local_git_repo
 
-* Args:
-    * `path`: **str**
-* Returns: **Repo**
+Creates a `repo` from the git repo at `path`.
 
-#### Repo.path(path)
-Gets the absolute to the file specified at `path` in the repo. Must be a relative path.
+```python
+def local_git_repo(path: str) -> Repo
+```
 
-* Args:
-  * `path`: **str**
-* Returns:  **localPath**
+#### Repo
 
-#### start_fast_build(dockerfile_path, img_name, entrypoint?)
-Builds a docker image.
+Represents a local code repository
 
-* Args:
-  * `dockerfile_path`: **str**
-  * `img_name`: **str**, e.g. blorgdev/backend or gcr.io/project-name/bucket-name
-  * `entrypoint?`: **str**
-* Returns: **Image**
+```python
+class Repo:
+  def path(path: str) -> localPath:
+    """Returns the absolute path to the file specified at path in the repo.
+    path must be a relative path.
+
+    Args:
+      path: relative path in repository
+    Returns:
+      A localPath resource, representing a local path on disk.
+    """
+```
+
+#### start_fast_build
 
-#### add(src, dest)
-Adds the content from `src` into the image at path `dest`. Paths must be relative.
+Initiates a docker image build that uses a cache and supports `add`s and `run`s.
 
-* Args:
-  * `src`: **localPath|gitRepo**
-  * `dest`: **str**
-* Returns: nothing
+TODO(dmiller): explain how this is fast, and image vs container builds?
+TODO(dmiller): explain the concept of the active build
+
+```python
+def start_fast_build(dockerfile_path: str, img_name: str, entrypoint: str = "") -> None
+```
+
+#### add
+
+Adds the content from `src` into the image at path `dest`.
+
+```python
+def add(src: Union[localPath, Repo], dest: str) -> None
+```
+
+#### run
 
-#### run(cmd, trigger?)
 Runs `cmd` as a build step in the image.
-If the `trigger` file is specified, the build step is only run if the file is changed. Path must be relative.
+If the `trigger` file is specified, the build step is only run if the file is changed.
 
-* Args:
-  * `cmd`: **str**
-  * `trigger?`: **List[str] | str**
-* Returns: nothing
+```python
+def run(cmd: str, trigger: Union[List[str], str] = []) -> None
+```
+
+#### k8s_service
 
-#### k8s_service(yaml_text, img)
 Creates a kubernetes service that tilt can deploy using the yaml text and the image passed in.
 
-* Args:
-  * `yaml_text`: **str** (text of yaml configuration)
-  * `img`: **Image**
-* Returns: **Service**
+```python
+def k8s_service(yaml_text: str, img: Image) -> Service
+```
+
+#### Image
+
+Represents a built Docker image
+
+```python
+class Image
+```
+
+#### composite_service
 
-#### composite_service(service_fns)
 Creates a composite service; tilt will deploy (and watch) all services returned by the functions in `service_fns`.
 
-* Args:
-  * `service_fns`: array of functions that each return **Service**
-* Returns: **Service**
+```python
+def composite_service(List[Callable[[], Service]]) -> Service
+```
+
+#### Service
```

Comment:
maybe put this def above `composite_service`?

Reply:
also aren't we no longe calling these "services"?

Reply:
Yeah it wasn't clear to me whether that was an internal rebranding our an external one as well. Happy to go through and make that change in a subsequent PR.